server/src/main/java/org/opensearch/indices/SystemIndices.java

```diff
@@ -63,7 +63,9 @@ public class SystemIndices {
 
     public SystemIndices(Map<String, Collection<SystemIndexDescriptor>> pluginAndModulesDescriptors) {
         SystemIndexRegistry.register(pluginAndModulesDescriptors);
-        this.runAutomaton = buildCharacterRunAutomaton(SystemIndexRegistry.SYSTEM_INDEX_DESCRIPTORS);
+        this.runAutomaton = buildCharacterRunAutomaton(
+            SystemIndexRegistry.SYSTEM_INDEX_DESCRIPTORS_MAP.values().stream().flatMap(Collection::stream).collect(Collectors.toList())
```

Comment:
Please extract this into a statcc method on `SystemIndexRegistry` class

Reply:
Extracted to a static method called getAllDescriptors and added a test